pkg/resource/cluster/pre_build_request.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cluster
+
+import (
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/pkg/errors"
+	"strconv"
+
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+// validateClusterNeedsUpdate this function's purpose is to requeue if the resource is currently unavailable and
+// to validate if resource update is required.
+func (rm *resourceManager) validateClusterNeedsUpdate(
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	// requeue if necessary
+	latestStatus := latest.ko.Status.Status
+	if latestStatus == nil || *latestStatus != "available" {
+		return nil, requeue.NeededAfter(
+			errors.New("Cluster cannot be updated as its status is not 'active'."),
+			requeue.DefaultRequeueAfterDuration)
+	}
+
+	// Set terminal condition when cluster is in create-failed state
+	if latestStatus != nil || *latestStatus == "create-failed" {
+		return nil, awserr.New("InvalidNodeStateFault", "Cluster is in create-failed state, cannot be updated.", nil)
```

Comment:
nit: Please don't create an `awserr` type. These are reserved from errors returned by the AWS API